resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java

```diff
@@ -338,6 +346,48 @@ public int compare(ResourceAddress addr1, ResourceAddress addr2) {
         
     }
 
+    private static final class ResourceAlternateAndTransportLocationComparator implements Comparator<ResourceAddress> {
+
+        @Override
+        public int compare(ResourceAddress addr1, ResourceAddress addr2) {
+
+            // for tcp
+            ResourceAddress alternate1 = addr1.getOption(ALTERNATE);
+            ResourceAddress alternate2 = addr2.getOption(ALTERNATE);
+
+            String alternate1IPv6 = alternate1 != null ? alternate1.getResource().toString() : null;
+            String alternate2IPv6 = alternate2 != null ? alternate2.getResource().toString() : null;
+
+            boolean alt1HasIPv6 = alternate1IPv6 != null ? alternate1IPv6.contains("[0:0:0:0:0:0:0:1]") : false;
+            boolean alt2HasIPv6 = alternate2IPv6 != null ? alternate2IPv6.contains("[0:0:0:0:0:0:0:1]") : false;
+
+            if (alt1HasIPv6 && alternate2 == null || alt2HasIPv6 && alternate1 == null) {
+                URI resource1 = addr1.getResource();
+                URI resource2 = addr2.getResource();
+                return resource1.compareTo(resource2);
```

Comment:
This is for comparing two TcpResourceAddresses such as:

[tcp://127.0.0.1:65535 (tcp://127.0.0.1:65535)]
with

[tcp://127.0.0.1:65535 (tcp://localhost:65535) socks/5+]
[tcp://[0:0:0:0:0:0:0:1]:65535 (tcp://localhost:65535) socks/5+]
Here we have alternate on the second one as IPv6, and no alternate on the first one, but the two should be identical, I think, so we do the comparison based only on resourceURI.